newfile.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type Book struct {
+	ID     string  `json:"id"`
+	ISBN   string  `json:"isbn"`
+	Title  string  `json:"title"`
+	Author *Author `json:"author"`
+}
+
+type Author struct {
+	Firstname string `json:"firstname"`
+	Lastname  string `json:"lastname"`
+}
+
+var books []Book
+
+func getBooks(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	w.Header().Set("Date", "Wed, 16 Aug 2022 12:15:38 GMT")
+	json.NewEncoder(w).Encode(books)
+}
+
+func getBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range books {
+		if item.ID == params["id"] {
+			json.NewEncoder(w).Encode(item)
+			return
+		}
+	}
+	// json.NewEncoder(w).Encode(&Book{})
+	fmt.Fprintf(w, "No Data available")
+}
+
+func createBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	var book Book
+	_ = json.NewDecoder(r.Body).Decode(&book)
+	book.ID = strconv.Itoa(rand.Intn((10000000)))
+	books = append(books, book)
+	json.NewEncoder(w).Encode(book)
+	fmt.Fprintf(w, "New Entry created and added to the database")
```

Comment:
Why we are using different response format in APIs, it would be better to stick with single response format for every API

Reply:
This is for adding a new entry. So , we will have to append it.

Reply:
We can have another type like Books that can hold all the books on which we can have method like All, Find, Create, Update, etc